lib/ocpp/v201/init_device_model_db.cpp

```diff
@@ -570,20 +492,32 @@ void InitDeviceModelDb::insert_attribute(const VariableAttribute& attribute, con
     if (insert_attributes_statement->step() != SQLITE_DONE) {
         throw InitDeviceModelDbError("Could not insert attribute: " + std::string(this->database->get_error_message()));
     }
+
+    const int64_t attribute_id = this->database->get_last_inserted_rowid();
+
+    if (attribute.value.has_value() ||
+        (attribute.type.has_value() && (attribute.type.value() == AttributeEnum::Actual) &&
+         default_actual_value.has_value())) {
```

Comment:
You could put this into a separate function so this reads a little better. You could also reusue it within L604